samplers/src/test/java/internal/CelBasedSamplerComponentProviderTest.java

```diff
@@ -0,0 +1,246 @@
+                + "CelBasedSamplingExpression{"
+                + "expression='\"example.com\" in attribute[\"http.response.header.host\"] && attribute[\"http.response.status_code\"] == 200', "
+                + "delegate=AlwaysOffSampler"
+                + "}, "
+                + "CelBasedSamplingExpression{"
+                + "expression='spanKind == \"SERVER\" && attribute[\"url.path\"].matches(\"/actuator.*\")', "
+                + "delegate=AlwaysOffSampler"
+                + "}, "
+                + "CelBasedSamplingExpression{"
+                + "expression='spanKind == \"SERVER\" && attribute[\"url.path\"].matches(\"/actuator.*\")', "
+                + "delegate=AlwaysOnSampler"
+                + "}"
+                + "]},"
+                + "remoteParentSampled:AlwaysOnSampler,"
+                + "remoteParentNotSampled:AlwaysOffSampler,"
+                + "localParentSampled:AlwaysOnSampler,"
+                + "localParentNotSampled:AlwaysOffSampler"
+                + "}");
+
+    // SERVER span to other path should be recorded and sampled
+    assertThat(
+            sampler.shouldSample(
+                Context.root(),
+                IdGenerator.random().generateTraceId(),
+                "GET /v1/users",
+                SpanKind.SERVER,
+                Attributes.builder()
+                    .put(HttpAttributes.HTTP_RESPONSE_STATUS_CODE, 200)
+                    .put(
+                        HttpAttributes.HTTP_RESPONSE_HEADER.getAttributeKey("host"),
+                        new String[] {"example.com"})
+                    .build(),
+                Collections.emptyList()))
+        .isEqualTo(SamplingResult.drop());
+
+    // SERVER span to /actuator.* path should be dropped. This is the first expression that matches.
+    assertThat(
+            sampler.shouldSample(
+                Context.root(),
+                IdGenerator.random().generateTraceId(),
+                "GET /actuator/health",
+                SpanKind.SERVER,
+                Attributes.builder().put("url.path", "/actuator/health").build(),
+                Collections.emptyList()))
+        .isEqualTo(SamplingResult.drop());
+
+    // SERVER span to other path should be recorded and sampled
+    assertThat(
+            sampler.shouldSample(
+                Context.root(),
+                IdGenerator.random().generateTraceId(),
+                "GET /v1/users",
+                SpanKind.SERVER,
+                Attributes.builder().put("url.path", "/v1/users").build(),
+                Collections.emptyList()))
+        .isEqualTo(recordAndSample());
+  }
+
+  static Sampler dropSampler(Sampler fallback, String... expressions) {
+    CelBasedSamplerBuilder builder = CelBasedSampler.builder(fallback);
+    for (String expression : expressions) {
+      try {
+        builder.drop(expression);
+      } catch (CelValidationException e) {
+        // Delegate to the provider to handle the exception
```

Comment:
would adding a `fail()` call in the `catch()` make this slightly more robust? If not, then we expect this to be hit sometime and should probably verify that. Otherwise, just let this method throw and the test can fail normally.

Reply:
Good catch. Was an easy fix.